### PR TITLE
OBJLoader: Better handle inconsistent face definitions.

### LINKED ISSUE #16211: OBJLoader: Loader does not return valid geometry with inconsistent face definitions
If the OBJ file specifies face indices inconsistently then the loader will return geometry with mismatched attribute lengths. I ran into this issue while loading an OBJ file I'm working with and it causes the geometry to not render. Unfortunately I can't share the model but I'll try to give a  rough example of what's happening.

The OBJ face list might be formed like so. Notice that one face does _not_ specify normals while the other faces do:
```obj
f 1050//4 1051//5 1049//6
f 185//7 183//8 179//9
f 500 505 1053
f 732//10 500//11 1053//12
f 153//13 144//14 137//15
```

This will result in a position attribute with a count of `15` and a normal array with a count of `12`, meaning the geometry will fail to render (and the normals and positions are now misaligned in their respective arrays). The problematic logic seems to be in [the `addFace` function](https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/OBJLoader.js#L279), which just ignores an empty normal values even if they're defined in sibling faces. You can see that this will also be a problem for UVs, as well.

The problem is not present in `OBJLoader2` and a mesh formed like this loads just fine in MeshLab, as well.

### PR DESCRIPTION
**Pull request by Mugen87**
Fixed #16211.

- If a face definition does not define normal indices, a face normal is computed.
- If a face definition does not define uv indices, `(0,0)` is set.
- Only adds normal or uv attribute if faces have respective (potentially inconsistent) definitions.

### CODE REVIEW

**Comment by Mugen87** [comment]
@gkjohnson Do you had something like this in mind?

File: examples/jsm/loaders/OBJLoader.js [hunk 1]
```diff
@@ -305,26 +346,45 @@ var OBJLoader = ( function () {
 				this.addVertex( ia, ib, ic );
 				this.addColor( ia, ib, ic );
 
+				// normals
+
+				if ( na !== undefined && na !== '' ) {
```
Comment (gkjohnson) [inline examples/jsm/loaders/OBJLoader.js:351]: It's probably not worth worrying about  until there's a model that demonstrates the issue but I wonder if it's possible for just _one_ of the vertices in a face to specify a normal while the other two do not.

**Comment by gkjohnson** [comment]
This looks good to me, thank you!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**